packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -71,69 +75,99 @@ class EnrollVerbHandler extends AbstractVerbHandler {
+  /// keys with clients.
+  ///
+  /// If the enrollment request originates from a CRAM authenticated connection:
+  ///
+  /// The enrollment is automatically approved and given privilege to the "__manage"
+  /// namespace group with "rw" access.
+  /// The default encryption private key and default self-encryption key are
+  /// securely stored in encrypted format within the keystore.
+  ///
+  /// If the enrollment request originates from an unauthenticated connection and
+  /// includes a valid OTP (One-Time Password), it is marked as pending.
+  ///
+  ///
+  /// The function returns a JSON-encoded string containing the enrollmentId
+  /// and its corresponding state.
+  ///
+  /// Throws "AtEnrollmentException", if the OTP provided is invalid.
   Future<void> _handleEnrollmentRequest(
-      HashMap<String, String?> verbParams,
+      EnrollParams enrollParams,
       currentAtSign,
       Map<dynamic, dynamic> responseJson,
       InboundConnection atConnection) async {
     if (!atConnection.getMetaData().isAuthenticated) {
-      var totp = verbParams['totp'];
-      if (totp == null || (await TotpVerbHandler.cache.get(totp)) == null) {
+      var totp = enrollParams.totp;
+      if (totp == null ||
+          (await TotpVerbHandler.cache.get(totp.toString()) == null)) {
         throw AtEnrollmentException(
             'invalid totp. Cannot process enroll request');
       }
     }
-    List<EnrollNamespace> enrollNamespaces = (verbParams['namespaces'] ?? '')
-        .split(';')
-        .map((namespace) =>
-            EnrollNamespace(namespace.split(',')[0], namespace.split(',')[1]))
-        .toList();
-    logger.finer('enrollNamespaces: $enrollNamespaces');
-
-    var enrollmentId = Uuid().v4();
-    var key = '$enrollmentId.$newEnrollmentKeyPattern.$enrollManageNamespace';
+    var enrollNamespaces = enrollParams.namespaces!;
+    var newEnrollmentId = Uuid().v4();
+    var key =
+        '$newEnrollmentId.$newEnrollmentKeyPattern.$enrollManageNamespace';
     logger.finer('key: $key$currentAtSign');
 
-    responseJson['enrollmentId'] = enrollmentId;
+    responseJson['enrollmentId'] = newEnrollmentId;
     final enrollmentValue = EnrollDataStoreValue(
         atConnection.getMetaData().sessionID!,
-        verbParams['appName']!,
-        verbParams['deviceName']!,
-        verbParams['apkamPublicKey']!);
+        enrollParams.appName!,
+        enrollParams.deviceName!,
+        enrollParams.apkamPublicKey!);
 
-    if (atConnection.getMetaData().isAuthenticated) {
-      // approve request from connection that are authenticated. This connection may be cram or legacyPkam authenticated
-      enrollNamespaces.add(EnrollNamespace(enrollManageNamespace, 'rw'));
+    if (atConnection.getMetaData().authType != null &&
+        atConnection.getMetaData().authType == AuthType.cram) {
+      // auto approve request from connection that is CRAM authenticated.
+      enrollNamespaces[enrollManageNamespace] = 'rw';
```

Comment:
Should we add another entry into `enrollNamespaces` like `"*":"rw"` ? How are we currently determining access permissions where the only entry in `enrollNamespaces` is `"$enrollManageNamespace":"rw"` ?

Reply:
we didn't handle this case. if the only access is __manage and keynamespace is .wavi,.buzz etc., then we are returning isAuthorized = false..We can add an entry *:rw for the privileged client to access all namespaces

Reply:
Has this been done? And is the `*:rw` namespace permission being checked and understood in the relevant places?

Reply:
yes. Made the relevant change in isAuthorized method to check for *:rw

Reply:
I can see there are tests for delete and update verbs for `*:rw` permissions but I don't see those tests for notify / scan / lookup / llookup  ...